gossip/src/cluster_info.rs

```diff
@@ -392,7 +384,6 @@ fn retain_staked(values: &mut Vec<CrdsValue>, stakes: &HashMap<Pubkey, u64>) {
             // the various dashboards.
             CrdsData::Version(_) => true,
             CrdsData::NodeInstance(_) => true,
-            // getHealth fails if account hashes are not propagated.
             CrdsData::AccountsHashes(_) => true,
```

Comment:
only question i have is related to my comment here: https://github.com/solana-labs/solana/pull/34979#discussion_r1470141737. Wouldn't setting this to `false` ensure this doesn't get propagated via `PullResponse`? Since `handle_pull_requests()` has:
```
if self.require_stake_for_gossip(stakes) {
    for resp in &mut pull_responses {
        retain_staked(resp, stakes);
    }
}
```
when generating a Pull Respons, which would always filter out the `AccountHashes`

Or is the issue with compatability? Because nodes that have not upgraded to this version will continue to always send all `AccountHashes` in their Pull Responses?

Reply:
> Or is the issue with compatability? Because nodes that have not upgraded to this version will continue to always send all AccountHashes in their Pull Responses?

yes, the issue is with nodes which don't have this code yet.
* your node sends the pull request to the cluster running v1.17.
* the remote node (which doesn't have this code yet) responds back with account-hashes.
* if it was `=> false` here, your node immediately throws them out and does not store in crds table.
* the next pull-request, everything repeats.

Reply:
cool got it. thank you.